WooCommerce/Classes/ViewRelated/Orders/Simple Payments/Method/SimplePaymentsMethodsViewModel.swift

```diff
@@ -103,20 +109,25 @@ final class SimplePaymentsMethodsViewModel: ObservableObject {
             guard let self = self else { return }
             self.showLoadingIndicator = false
 
-            if error == nil {
-                onSuccess()
-                self.presentNoticeSubject.send(.completed)
-            } else {
+            if let error = error {
                 self.presentNoticeSubject.send(.error(Localization.markAsPaidError))
+                self.trackFlowFailed()
+                return DDLogError("⛔️ Error updating simple payments order: \(error)")
             }
-            // TODO: Analytics
+
+            onSuccess()
+            self.presentNoticeSubject.send(.completed)
+            self.trackFlowCompleted(method: .cash)
         }
         stores.dispatch(action)
+        trackCollectIntention(method: .cash)
```

Comment:
Because this is part of `markOrderAsPaid` the event only fires if I tap the button on the alert to collect payment (after tapping on the cash row). That means if I select cash, and then cancel and select card, only the card event is fired.

In contrast, the same event for card payments fires as soon as I tap the card row, even if the card payment isn't successful and I end up switching to cash payment. (So if I select card, and then cancel and select cash, I end up with both events fired.)

Should we move this so we track the collect intention at the same point for both payment methods?

Reply:
Nice catch, I'll move it to when the user initially taps the row to have the same behavior in both scenarios!

Reply:
Fixed https://github.com/woocommerce/woocommerce-ios/pull/5633/commits/a07f1bcc016a2aff0967c30902e350c4ba9005d1